app/javascript/mastodon/emoji_data_light.js

```diff
@@ -0,0 +1,17 @@
+// @preval
+const data = require('emoji-mart/dist/data').default;
+const pick = require('lodash/pick');
+
+const condensedEmojis = {};
+Object.keys(data.emojis).forEach(key => {
+  condensedEmojis[key] = pick(data.emojis[key], ['short_names', 'unified', 'search']);
+});
+
+// JSON.parse/stringify is to emulate what @preval is doing and avoid any
+// inconsistent behavior in dev mode
+module.exports = JSON.parse(JSON.stringify({
+  emojis: condensedEmojis,
+  skins: data.skins,
+  categories: data.categories,
```

Comment:
Since this is only used in `emoji_index_light`, maybe we should use one-letter keys to save space?

Reply:
In `emojione_light` we used arrays to get rid of the keys altogether.

Reply:
Yeah we could maybe trim this even further. Using arrays instead of objects may be a good start. I'm interested though in the tradeoff between transfer cost and decompression cost – e.g. with arrays instead of objects, how much more expensive is it to decompress on the client side? Does it offset the gains in transfer/parse costs?

The search code could also be trimmed, because there are lots of emoji-mart options we're not using. I also believe there may be repeated information between `emojione_light.js` and `emoji_data_light.js`.

Reply:
> I'm interested though in the tradeoff between transfer cost and decompression cost – e.g. with arrays instead of objects, how much more expensive is it to decompress on the client side? Does it offset the gains in transfer/parse costs?

I'm curious, how would you measure that? On the surface, I would expect decompression to be faster since the payload is smaller and arrays are a somehow simpler data structure than hash-maps.

Reply:
| I also believe there may be repeated information between emojione_light.js and `emoji_data_light.js`.

How come `emojione_light` is still needed/used?

Reply:
Good question. I think it may have something to do with how the emoji codepoints are expressed in Twemoji? @Gargron?